lib/statsd.js

```diff
@@ -526,3 +502,59 @@ Client.prototype.childClient = function (options) {
 
 exports = module.exports = Client;
 exports.StatsD = Client;
+
+function udsErrorHandler(client, lastSocketCreateTime) {
```

Comment:
1. I'm not sure if you want these handlers here at the end
1. I'm not sure if you'd prefer `client` be `this` and we just put these helpers on the `Client` prototype

Reply:
I think it's fine as is.  Great to see the nice docs there.